### PR TITLE
handshake compression + compressData viper config + adapt tests

### PR DESCRIPTION
**Pull request by traysh**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -377,6 +381,18 @@ func hbdEncode(heartbeatTimeout time.Duration, packetEncoder codec.PacketEncoder
...
+	if dataCompression {
+		compressedData, err := compression.DeflateData(data)
+		if err != nil {
+			panic(err)
+		}
+
+		if len(compressedData) < len(data) {
+			data = compressedData
+		}
+
```
Comment (andrehp) [inline agent/agent.go]: Remove this empty line

File: examples/demo/cluster/main.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func configureBackend() {
 }
 
 func configureFrontend(port int) {
-	ws := acceptor.NewWSAcceptor(fmt.Sprintf(":%d", port))
+	ws := acceptor.NewTCPAcceptor(fmt.Sprintf(":%d", port))
```
Comment (andrehp) [inline examples/demo/cluster/main.go]: Shouldn't this still be WS?

Comment (traysh) [reply examples/demo/cluster/main.go]: Yes, will fix it

File: agent/agent.go (hunk 1 above)
Comment (traysh) [reply agent/agent.go]: done

**Pull request merged**